controllers/Router.php

```diff
@@ -6,40 +6,41 @@
 use GuzzleHttp\Psr7\ServerRequest;
 use GuzzleHttp\Psr7\Uri;
 use GuzzleHttp\Psr7\UriNormalizer;
-use OC_Blog\Config\ConstantGlobal;
+use OC_Blog\Tools\ConstantGlobal;
 
 
 class Router {
 
-
-	public function run($twig){
+	/**
+	 * Décompose l'url et instancie le controller demandé.
+	 *
+	 * @param object $twig
+	 */
+	public function run(object $twig): void {
 
 		$url = ( new ConstantGlobal(ServerRequest::fromGlobals()) )->getServerUri();
-		$url = $this->cleanUri($url);
-
-		$this->routeRequest($twig, $url);
-	}
+		$uri = $this->cleanUri($url);
 
-	public function routeRequest( $twig, $uri ) {
 		try {
 			$url = explode( '/', filter_var( $uri, FILTER_SANITIZE_URL ) );
+			$slug = $this->slug($url);
+
 			if ( isset($url)) {
 
 				$controller      = ucfirst( strtolower( $url[1] ) );
 				$controllerClass = "OC_Blog\\Controllers\\Controller" . $controller;
+				$method = $this->getMethod($url, $controller);
 
-				if ( class_exists( $controllerClass)  && empty($url[2])) {
-					new $controllerClass($twig);
-				}elseif (class_exists( $controllerClass)  && !empty($_POST)){
+				if (class_exists( $controllerClass)  && !empty($_POST)){
```

Comment:
Codacy found an issue: [Direct use of $_POST Superglobal detected.](https://app.codacy.com/manual/PatOpen/OC_Blog/pullRequest?prid=6061935)